Makefile

```diff
@@ -501,9 +501,19 @@ _ensure-kong-system-namespace:
 # TODO: In order not to rely on 'main' version of Gateway API CRDs address but
 # on the tag that is used in code (defined in go.mod) address this by solving
 # https://github.com/Kong/gateway-operator/pull/480.
+# NOTE: the kubeconfig context gets changed to kgo, a new context that
+# impersonates KGO service accounts. This is necessary to test RBACs are set up correctly.
+# In case you want to run other kubectl commands with admin grants from other terminals, you may want 
+# to duplicate your kubeconfig to use different contexts.
+# Once the operator is stopped, the context will be switched back to the original one.
 .PHONY: run
-run: webhook-certs-dir manifests generate install.all _ensure-kong-system-namespace
-	@$(MAKE) _run
+run: webhook-certs-dir manifests generate install.all install.rbacs _ensure-kong-system-namespace
+	@(CONTEXT=$$(kubectl config current-context); \
+	trap 'kubectl config use-context $$CONTEXT; exit 1' INT ERR EXIT; \
+	kubectl config set-credentials kgo --token=$(shell kubectl create token --namespace=kong-system controller-manager); \
+	kubectl config set-context kgo --cluster=$(shell kubectl config get-contexts | grep '^\*' | tr -s ' ' | cut -d ' ' -f 3) --user=kgo --namespace=kong-system; \
+	kubectl config use-context kgo; \
```

Comment:
Instead of manipulating current context how about we just run the operator using a curated context? We could specify e.g. the cli arg `kubeconfig` / env `GATEWAY_OPERATOR_KUBECONFIG` to a kubeconfig stored somewhere in `/tmp`. WDYT?

This will not require us to change the context if we'd like to apply objects against the cluster in a separate terminal.